repo/test/test_repo.go

```diff
@@ -97,15 +97,49 @@ func NewTestRepo(rc *regclient.Client) (mr *repo.MemRepo, err error) {
 		if err != nil {
 			return nil, err
 		}
-
-		if err = createDataset(mr, tc); err != nil {
+		if _, err := createDataset(mr, tc); err != nil {
 			return nil, fmt.Errorf("%s error creating dataset: %s", k, err.Error())
 		}
 	}
 
 	return
 }
 
+// NewTestRepoForLog generates a repository with a dataset that has a history, usable for testing purposes
+func NewTestRepoForLog(rc *regclient.Client) (mr *repo.MemRepo, refs []repo.DatasetRef, err error) {
```

Comment:
I think we need to have a larger conversation about making datasets with long histories, and repos that have datasets with long histories. That time isn't now. For now, can we rename this to `NewTestRepoWithHistory`?

Reply:
jesus that name is so much better

and yes this is definitely not the optimal way to do this